server/methods/messageSearch.js

```diff
@@ -25,7 +25,7 @@ Meteor.methods({
 			if (!Meteor.call('canAccessRoom', rid, currentUserId)) {
 				return result;
 			}
-		} else if (RocketChat.settings.get('Message_GlobalSearch') !== true) {
```

Comment:
looks like we don't need the old setting any more right? if so, I would ask to remove `Message_GlobalSearch`

Reply:
You mean that is should be removed from the config?https://github.com/RocketChat/Rocket.Chat/blob/ee06a917a287d8158dd6f9cc5f67acab79ec061c/packages/rocketchat-lib/server/startup/settings.js#L1372

Reply:
yes. a migration will also be needed to reflect the current value of `Message_GlobalSearch` into the new `Search.defaultProvider.GlobalSearchEnabled` setting.

additionally just found that old templates are also not being used anymore, so we would need to also delete the files:

- [messageSearch.html](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-ui-flextab/client/tabs/messageSearch.html)
- [messageSearch.js](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-ui-flextab/client/tabs/messageSearch.js)

and remove its references at:

- [rocketchat-ui-flextab/package.js#L24](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-ui-flextab/package.js#L24)
- [rocketchat-ui-flextab/package.js#L31](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-ui-flextab/package.js#L31)

Reply:
Fully agree: @sampaiodiego Is there any standard way for setting migration or do we need to do it programatically (in our code).

Reply:
the migration is simple.. take a look at migration [112](https://github.com/RocketChat/Rocket.Chat/blob/pr-search/server/startup/migrations/v112.js#L7) .. you just need to `find` the old setting, grab its value and run an `upsert` (the settings might have been created yet, so an `upsert` is more recommended for this case) on the new setting with the old setting's value. after that you can `remove` the old setting.

Reply:
Cool, i will commit this as soon it is ready.

Reply:
@sampaiodiego I added the migration. Can you please check? Should do the trick right?

Reply:
you forgot to remove the setting :) I [just did it](https://github.com/redlink-gmbh/Rocket.Chat/commit/7acd77072188168e27841919be9c3514cd1f48fa)

👍